BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/ble/beacon/Beacon.java

```diff
@@ -140,8 +141,12 @@ public boolean hasBeenSeenSince(long timestamp) {
     public float getDistance() {
         List<AdvertisingPacket> recentAdvertisingPackets = (List<AdvertisingPacket>) getAdvertisingPacketsFromLast(5, TimeUnit.SECONDS);
         int[] recentRssis = AdvertisingPacketUtil.getRssisFromAdvertisingPackets(recentAdvertisingPackets);
-        float meanRssi = AdvertisingPacketUtil.getMeanRssi(recentRssis);
-        return BeaconDistanceCalculator.calculateDistanceTo(this, meanRssi);
+        RssiArmaFilter filter = new RssiArmaFilter();
+        filter.addMeasurement((int) AdvertisingPacketUtil.getMeanRssi(recentRssis),recentRssis.length);
+        float filteredRssi = (float) filter.getFilteredRssi();
```

Comment:
Why is it necessary to convert? Wouldnt it make more sense to stick to either `float` or `double` in both the `RSSSIArmaFilter` as well as in the distanceCalculation.